spec/firebase_app_distribution_api_client_spec.rb

```diff
@@ -631,11 +631,255 @@
+        displayName: display_name
+      }
+      stubs.post("/v1/projects/#{project_number}/groups?groupId=#{group_alias}", payload.to_json, headers) do |env|
+        [
+          200,
+          {}, # response headers
+          {} # response body
+        ]
+      end
+
+      result = api_client.create_group(project_number, group_alias, display_name)
+
+      expect(result.success?).to eq(true)
+    end
+
+    it 'is successful when the group already exists' do
+      payload = {
+        name: "projects/#{project_number}/groups/#{group_alias}",
+        displayName: display_name
+      }
+      stubs.post("/v1/projects/#{project_number}/groups?groupId=#{group_alias}", payload.to_json, headers) do |env|
+        [
+          409,
+          {}, # response headers
+          {} # response body
+        ]
+      end
+
+      result = api_client.create_group(project_number, group_alias, display_name)
+
+      expect(result).to eq(true)
+    end
+
+    it 'fails and prints correct error message for a 400' do
+      display_name = "Group display name that is longer than the maximum allowed length for a group name"
+      payload = { name: "projects/#{project_number}/groups/#{group_alias}",
+                  displayName: display_name }
+      stubs.post("/v1/projects/#{project_number}/groups?groupId=#{group_alias}", payload.to_json, headers) do |env|
+        [
+          400,
+          {}, # response headers
+          {} # response body
+        ]
+      end
+
+      expect { api_client.create_group(project_number, group_alias, display_name) }
+        .to raise_error(ErrorMessage::INVALID_TESTER_GROUP_NAME)
+    end
+
+    it 'fails and prints correct error message for a 404' do
+      bad_project_number = "bad_project_number"
+      payload = { name: "projects/#{bad_project_number}/groups/#{group_alias}",
+                  displayName: display_name }
+      stubs.post("/v1/projects/#{bad_project_number}/groups?groupId=#{group_alias}", payload.to_json, headers) do |env|
+        [
+          404,
+          {}, # response headers
+          {} # response body
+        ]
+      end
+      expect { api_client.create_group(bad_project_number, group_alias, display_name) }
+        .to raise_error(ErrorMessage::INVALID_PROJECT)
+    end
+  end
+
+  describe '#remove_tester_group' do
+    let(:headers) do
+      {
+        'Authorization' => 'Bearer auth_token',
+        'Content-Type' => 'application/json',
+        'X-Client-Version' => "fastlane/#{Fastlane::FirebaseAppDistribution::VERSION}"
+      }
+    end
+    let(:project_number) { "project_number" }
+    let(:group_alias) { "group_alias" }
+
+    it 'is successful' do
+      stubs.delete("/v1/projects/#{project_number}/groups/#{group_alias}", headers) do |env|
+        [
+          200,
+          {}, # response headers
+          {} # response body
+        ]
+      end
+
+      result = api_client.delete_group(project_number, group_alias)
```

Comment:
Consider returning the body of the response, which would actually be null, and verifying that in the test instead of returning the response wrapper object.